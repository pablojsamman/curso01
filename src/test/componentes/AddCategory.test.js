import React from 'react'
import '@testing-library/jest-dom'

import { shallow } from "enzyme"
import AddCategory from "../../Componentes/AddCategory"

describe('pruebas en el componente addcategory', () => {
    
    const setCategorias = jest.fn()
    let wrapper = shallow(<AddCategory setCategorias={setCategorias} />)

    beforeEach( () => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategorias={setCategorias} />)
    })

    test('debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe cambiar input', () => {

        const input = wrapper.find('input')
        const value = 'Hola Mundo'
        input.simulate('change', { target: { value } })
        
    })
    
    test('no debe de llamar la informacion', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect(setCategorias).not.toHaveBeenCalled()
        
    })

    test('setcategorias y limpia input', () => {

        const value = 'Hola Mundo'
        wrapper.find('input').simulate('change', { target: { value } })

        wrapper.find('form').simulate('submit', { preventDefault(){} })
        
        expect(setCategorias).toHaveBeenCalled()
        expect(setCategorias).toHaveBeenCalledWith(expect.any(Function))

        expect(wrapper.find('input').prop('value')).toBe('')
        
    })
    

})
