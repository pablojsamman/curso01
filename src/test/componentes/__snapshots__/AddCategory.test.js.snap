// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente addcategory debe mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
